docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md

```diff
@@ -0,0 +1,416 @@
+
+## Trailing commas
+
+During deserialization, `Newtonsoft.Json` ignores trailing commas by default. In some cases, it ignores multiple trailing commas (for example, `[{"Color":"Red"},{"Color":"Green"},,]`). The `System.Text.Json` default is to throw exceptions for trailing commas because the [RFC 8259](https://tools.ietf.org/html/rfc8259) specification doesn't include them. For information about how to allow trailing commas, see [Allow comments and trailing commas](system-text-json-how-to.md#allow-comments-and-trailing-commas). There's no way to allow multiple trailing commas.
+
+## Converter registration precedence
+
+The `Newtonsoft.Json` registration precedence for custom converters is as follows:
+
+* Attribute on property
+* Attribute on type
+* `Converters` collection
+
+This order means that a custom converter in the `Converters` collection is overridden by a converter that is registered by applying an attribute at the type level. Both of those registrations are overridden by an attribute at the property level.
+
+The `System.Text.Json` registration precedence for custom converters is different:
+
+* Attribute on property
+* `Converters` collection
+* Attribute on type
+
+The difference here is that a custom converter in the `Converters` collection overrides an attribute at the type level. The intention behind this order of precedence is to make run-time changes override design-time choices. There's no way to change the precedence.
+
+## `Newtonsoft.Json` [JsonProperty] attribute
+
+`Newtonsoft.Json` has a `[JsonProperty]` attribute that is used for a variety of functions. `System.Text.Json` has single-purpose attributes instead of a multi-purpose attribute. The following attributes serve purposes similar to some of the `Newtonsoft.Json` attribute functions:
+
+* [[JsonPropertyName]](system-text-json-how-to.md#customize-individual-property-names)
+* [[JsonConverter]](system-text-json-converters-how-to.md#register-a-custom-converter)
+
+Other functions of the `Newtonsoft.Json` `[JsonProperty]` attribute have no equivalent attribute in `System.Text.Json`. For some of those functions, there are other ways to accomplish the same purpose, as detailed in the following sections of this article.
+
+## Character escaping
+
+During serialization, `Newtonsoft.Json` is relatively permissive about letting characters through without escaping them. (That is, it doesn't replace them with `\uxxxx` where `xxxx` is the Unicode code of the character.) `System.Text.Json` escapes more characters by default to provide defense-in-depth protections against cross-site scripting (XSS) or information disclosure attacks. `System.Text.Json` escapes all non-ASCII characters by default, so you don't need to do anything if you're using `StringEscapeHandling.EscapeNonAscii`. For information about how to override the default `System.Text.Json` behavior, see [Customize character encoding](system-text-json-how-to.md#customize-character-encoding).
+
```

Comment:
Another default behavior that is worth mentioning is deserialization of `System.Object` properties in POCOs and including `Dictionary<string, object>`.

For "primitive" values, Newtonsoft.Json "guesses" the type and stores string/long/double/boolean as a boxed `object` when deserializing, and `JObject` for "complex" values in the JSON payload. It stores `null` when the payload has "null" in it.

S.T.Json stores a boxed `JsonElement` for "primitive" and "complex" values within the `System.Object` property or dictionary value. It stores `null` when the payload has "null" in it.

Here's a potential workaround (at least for the primitives):
https://github.com/dotnet/corefx/issues/38713#issuecomment-535166692

Reply:
I have those in the workaround section rather than here because these sections were intended to be differences in default behavior that no workaround was necessary for, i.e., built-in functionality could get you to Newtonsoft.Json behavior.   To make this now a more general default-differences section, I'll add sections for these and point forward to the workaround how-to sections.

Reply:
On second thought I'll just bring the workaround back to this section and eliminate the other section.